From: Stefan Haberland <stefan.haberland@de.ibm.com>
Subject: s390/dasd: prevent incorrect length error under z/VM after PAV changes
Patch-mainline: v4.5-rc5
Git-commit: 020bf042e5b397479c1174081b935d0ff15d1a64
References: bnc#968500, LTC#136670

Description:  dasd: prevent incorrect length error under z/VM
Symptom:      DASD driver error messages signaling incorrect length
              error after LCU changes.
Problem:      The channel checks the specified length and the provided
              amount of data for CCWs and provides an incorrect length
              error if the size does not match. Under z/VM with
              simulation activated the length may get changed for read
              unit address configuration requests which are issued
              after changes to the LCU are detected.
Solution:     Set the suppress length indication bit for read unit
              address configuration CCWs. This is stated as
              good CCW coding practice and avoids errors under z/VM.
Reproduction: Change the PAV setup on a LCU and have Linux running
              under z/VM.

Upstream-Description:

              s390/dasd: prevent incorrect length error under z/VM after PAV changes

              The channel checks the specified length and the provided amount of
              data for CCWs and provides an incorrect length error if the size does
              not match. Under z/VM with simulation activated the length may get
              changed. Having the suppress length indication bit set is stated as
              good CCW coding practice and avoids errors under z/VM.

              Cc: stable@vger.kernel.org
              Signed-off-by: Stefan Haberland <stefan.haberland@de.ibm.com>
              Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Stefan Haberland <stefan.haberland@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/block/dasd_alias.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/s390/block/dasd_alias.c
+++ b/drivers/s390/block/dasd_alias.c
@@ -698,7 +698,7 @@ static int reset_summary_unit_check(stru
 	ASCEBC((char *) &cqr->magic, 4);
 	ccw = cqr->cpaddr;
 	ccw->cmd_code = DASD_ECKD_CCW_RSCK;
-	ccw->flags = 0 ;
+	ccw->flags = CCW_FLAG_SLI;
 	ccw->count = 16;
 	ccw->cda = (__u32)(addr_t) cqr->data;
 	((char *)cqr->data)[0] = reason;
